include/jni/functions.hpp

```diff
@@ -589,7 +589,7 @@ namespace jni
     inline UniqueEnv AttachCurrentThread(JavaVM& vm)
        {
         JNIEnv* result;
-        CheckErrorCode(vm.AttachCurrentThread(&result, nullptr));
+        CheckErrorCode(vm.AttachCurrentThread(reinterpret_cast<void**>(&result), nullptr));
```

Comment:
https://github.com/daedric/jni.hpp/commit/ef2b450e381b991bd4498e0158e89bf0d200b9be this would be my solution without template :)